tiledb/sm/filesystem/azure.cc

```diff
@@ -173,6 +175,27 @@ Status Azure::init(const Config& config, ThreadPool* const thread_pool) {
   options.Retry.RetryDelay = retry_delay_;
   options.Retry.MaxRetryDelay = max_retry_delay;
 
+  ::Azure::Core::Http::CurlTransportOptions transport_opts;
+  SSLConfig ssl_cfg = SSLConfig(config);
+
+  if (!ssl_cfg.ca_file().empty()) {
+    transport_opts.CAInfo = ssl_cfg.ca_file();
+  }
+
+  if (!ssl_cfg.ca_path().empty()) {
+    LOG_WARN(
+        "Azure ignores the `ssl.ca_path` configuration key, "
+        "use `ssl.ca_file` instead");
+  }
+
+  if (ssl_cfg.verify() == false) {
+    transport_opts.SslVerifyPeer = false;
+  }
+
+  auto transport =
+      make_shared<::Azure::Core::Http::CurlTransport>(HERE(), transport_opts);
```

Comment:
This will cause the Curl transport to be used on Windows, which uses WinHTTP by default. Not sure if we care though.

Reply:
It looks like the WinHTTPTransportOptions has two settings `IgnoreUnknownCertificateAuthority` and `IgnoreInvalidCertificateCommonName` which we could disable with `ssl.verify = false`. However, there's no option for specifying a certificate chain so it'd end up like GCS in that we only have the one switch on Windows.

There's also `TransportOptions::DisableTlsCertificateValidation` that is the more generic interface that we could use instead so that we don't care at all what Azure uses behind the curtain which seems slightly safer.

Reply:
I take that back. If we don't use the curl transport here we'll not be able to specify a ca_file option on Linux which defeats the entire point of the PR (at least for Azure). So we can either leave it as it is, or we can do a `if constexpr` to pick the curl transport on not Windows and use the default version on Windows? Given it works fine on Windows as is, I'm inclined to leave it.

Reply:
Windows not honoring the `ca_file` isn't really an issue since we're really only concerned that it works on Linux, specifically in Docker containers that might not have the ca_bundle package installed.

@ihnorton Have you got any good arguments for or against libcurl on Windows?

Reply:
I guess lean toward WinHTTP, but not a strong preference

Reply:
I've updated the Azure adapter to use WinHTTP on Windows and the cURL adapter on !Windows. I ended up having to use the preprocessor though since the WinHTTP approach ends up attempting to include winhttp.h which makes things not work on !Windows.